### ISSUE TITLE
Deploy uploads corrupt zip

### ISSUE THREAD

**Issue by brianzinn** [google-github-actions/deploy-cloud-functions#41]
### TL;DR
The action appears to not fully upload the zip (it's a corrupt zip - manually extracting gives errors.  Viewing the function provides a link to the build errors.

**Expected behavior**
Function should deploy without errors.

**Observed behavior**
This fails 3 times and the build fails:
```bash
End-of-central-directory signature not found. Either this file is not
a zipfile, or it constitutes one disk of a multi-part archive. In the
latter case the central directory and zipfile comment will be found on
the last disk(s) of this archive.
```

Downloading the .zip file from the function in console there are errors unzipping locally.

### Reproduction

**Action YAML**
Due to security and repeatability - we only reference versioned external actions.  Assuming that is the same as `@main`.
```yaml
       - id: actual-deploy
         uses: google-github-actions/deploy-cloud-functions@v0.2.0
         with:
           name: <redacted>
           runtime: nodejs10
           credentials: ${{ secrets.GCP_SA_KEY }}
           entry_point: <redacted>
           timeout: 240
```

This is a fully working YAML
```yaml
jobs:
  release:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: install build
        run: |
          yarn install
          yarn build
      - name: Set up Cloud SDK
        uses: google-github-actions/setup-gcloud@master
        with:
          project_id: ${{ secrets.GCP_PROJECT_ID }}
          service_account_key: ${{ secrets.GCP_SA_KEY }}
          export_default_credentials: true
      - name: Deploy Function <redacted>
        run: gcloud functions deploy <redacted> --runtime nodejs10 --trigger-http  --entry-point <redacted> --timeout 240 --set-env-vars CLOUD_SQL_CONNECTION_NAME=${{ secrets.CLOUD_SQL_CONNECTION_NAME }}
  
```

**Additional information**
Nothing atypical.  It works with regular gcloud command.

**Comment by bharathkkb** [comment]
Thanks for the report @brianzinn 
If the repo where this occurred is public, could you link it? It would be super helpful to repro

**Comment by brianzinn** [comment]
it's a private repo unfortunately.  I'll share more log details:
```bash
Fetching storage object: gs://gcf-sources-<redacted>-us-central1/<function-name>-<redacted>/version-4/function-source.zip#<redacted>
Copying gs://gcf-sources-<redacted>-us-central1/<function-name>-<redacted>/version-4/function-source.zip#<redacted>
/ [0 files][ 0.0 B/271.4 KiB] / [1 files][271.4 KiB/271.4 KiB]
Operation completed over 1 objects/271.4 KiB.
Archive: /tmp/source-archive.zip
End-of-central-directory signature not found. Either this file is not
a zipfile, or it constitutes one disk of a multi-part archive. In the
latter case the central directory and zipfile comment will be found on
the last disk(s) of this archive.
unzip: cannot find zipfile directory in one of /tmp/source-archive.zip or
/tmp/source-archive.zip.zip, and cannot find /tmp/source-archive.zip.ZIP, period.
ERROR
ERROR: error fetching storage source: generic::unknown: retry budget exhausted (3 attempts): fetching gcs source: unpacking source from gcs: source fetch container exited with non-zero status: 9
```

The code in my repo is not proprietary and there are no secrets in it, but I had to add some redactions, because you are publishing files I exclude from `.gcloudignore` to the zip.  The other weird thing was that the corrupt zip file that I downloaded contained a zip file inside of it named similar to 'cs-0041.zip', which was also corrupt, but did contain most of the same files.  So, I was able to unzip those files as well.

I can email you the full log details and function versions as per the logs.  Unfortunately I have deleted the original zips, but if you tell me how to download them I can send them to you as well.  I am unable in the cloud console to select the older versions to download.

**Comment by brianzinn** [comment]
@bharathkkb got a public reproduction setup for you.  Needed to sanitize the code a bunch to make it public.  Hope it helps.

This is the output from the action.
```bash
Updating function deployment
Updating function deployment
Function deployment updated
No URL set. Only HttpsTrigger Cloud Functions have URL.
```

My deployed function details are:
`Deployment failure: Build failed: Build error details not available. Please check the logs at https://console.cloud.google.com/logs/viewer?project=...`

It is the same build error as above, but you should be able to reproduce now.  Here is the commit that broke the build:
https://github.com/brianzinn/easypost-webhooks-serverless/commit/2dc7bd84434a2de2f63175a459d1c486f00719cf

also note that before the referenced commit that it did deploy successfully.  cheers.

**Comment by bharathkkb** [comment]
Thanks for the repo @brianzinn 
I was able to repro the error and have a possible fix for this in #48

**Comment by bharathkkb** [comment]
Should be fixed by https://github.com/google-github-actions/deploy-cloud-functions/releases/tag/v0.4.0, feel free to reopen if it still occurs
